tensorcircuit/backends/jax_backend.py

```diff
@@ -630,7 +631,7 @@ def is_sparse(self, a: Tensor) -> bool:
         return isinstance(a, sparse.BCOO)  # type: ignore
 
     def device(self, a: Tensor) -> str:
-        dev = a.device()
+        (dev,) = a.devices()
```

Comment:
does `array.devices` work for older versions of jax? i.e. what about the backward compatibility

Reply:
also `device` is not only a deprecation, it was removed since jax 0.4.27 according to its changelog

Reply:
tested as early as jax0.4.6 (March 2023), and `array.devices()` works in that version